### DISCUSSION TITLE
Is it possible to modify an existing gomponent?

### DISCUSSION THREAD

**Discussion by AxelRHD** [maragudk/gomponents#191] [Q&A] [answered]
Is it possible to modify an already created component (adding new childs, chaning atttributes, etc.)?

Thank you!

**Comment by markuswustenberg** [comment]
Hi @AxelRHD . Generally, the style of gomponents as seen in the examples is quite declarative, so you'd have to build your own components in a more imperative style if you need that. All you need to do is implement the `gomponents.Node` interface, and you can use your components everywhere, also with existing helpers and functions.

I hope that answers your question? 😊

**Reply by AxelRHD** [reply]
I hope so. I am just a seasoned programmer. Maybe you can give a short example?

**Reply by markuswustenberg** [reply, marked as answer]
You could do something like:

```go

type YourComponent struct {
    children []g.Node
}

func (c *YourComponent) AddChild(child Node) {
    c.children = append(c.children, child)
}

func (c *YourComponent) Render(w io.Writer) error {
    // Render however you want to the writer
    return nil
}
```

**Reply by AxelRHD** [reply]
I see! Children is a slice of the included objects. I think, that will do it.

Thanks a lot!
